### DISCUSSION TITLE
DuckDB GIS viewer

### DISCUSSION THREAD

**Discussion by yuchuanjin** [dbeaver/dbeaver#21491] [Q&A]
Does GIS viewer not support DuckDB? I find that you can generate geometry data format when querying, which looks like same with other databases like Postgres. Is there any way to visualize the geo data on the map?

**Comment by LonwoLonwo** [comment]
Hello @yuchuanjin 

You can find instructions here: https://github.com/dbeaver/dbeaver/wiki/Working-with-Spatial-GIS-data#viewing-string-or-binary-data-from-any-database-on-a-map

**Reply by yuchuanjin** [reply]
I can't find that option. I use spatial extension in duckdb and use ST_POINT to generate the geometry point. The data type is GEOMETRY. What should I do for this?
![image](https://github.com/dbeaver/dbeaver/assets/145332028/2ee0082f-6069-4985-8cef-305406101553)
![image](https://github.com/dbeaver/dbeaver/assets/145332028/692284ba-1f9b-4e92-96a6-8df50b4dcdae)

**Reply by yuchuanjin** [reply]
I can convert it to string to get that option. But it's already a GEOMETRY type, can't it display directly on the map?

**Reply by LonwoLonwo** [reply]
Ok, I have a workaround for you.

1. Create a new virtual column as String as a copy of your geometry column (just use a column name as an expression)

![2023-10-12 09_33_59-DBeaver Ultimate 23 2 2 - _duck_090 db_ Script-144](https://github.com/dbeaver/dbeaver/assets/45152336/9f478f0e-dd3c-4c7c-8b1b-5f6e68fc3183)

![2023-10-12 09_34_32-DBeaver Ultimate 23 2 2 - _duck_090 db_ Script-144](https://github.com/dbeaver/dbeaver/assets/45152336/e4fe2fa7-9203-4e5f-9bcb-5d66c9e59e8a)

2. And after that - transform it to geometry.

![2023-10-12 09_35_15-DBeaver Ultimate 23 2 2 - _duck_090 db_ Script-144](https://github.com/dbeaver/dbeaver/assets/45152336/07529a93-f562-45b0-bba4-c54fd5e02f1e)

**Reply by yuchuanjin** [reply]
That's good, thanks for the help!
